### ISSUE TITLE
Possible to share your code of training nerfdet on arkit?

### ISSUE THREAD

**Issue by Pixie8888** [SerCharles/CN-RMA#2]
Dear author,

Do you mind sharing your code of training nerfdet on arkit? I am new to 3d object detection and didn't know how to implement it.

**Comment by SerCharles** [comment]
model = dict(
    type='nerfdet',
    pretrained='torchvision://resnet50',
    backbone=dict(
        type='ResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        norm_cfg=dict(type='BN', requires_grad=False),
        norm_eval=True,
        style='pytorch'),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        num_outs=4),
    neck_3d=dict(
        type='FastIndoorImVoxelNeck',
        in_channels=256,
        out_channels=128,
        n_blocks=[1, 1, 1]),
    bbox_head=dict(
        type='SunRgbdImVoxelHeadV2',
        n_classes=17,
        n_channels=128,
        n_reg_outs=7,
        n_scales=3,
        limit=27,
        centerness_topk=18),
    voxel_size=(.16, .16, .2),
    n_voxels=(40, 40, 16),
    aabb=([-2.7, -2.7, -0.78], [3.7, 3.7, 1.78]),
    near_far_range=[0.2, 8.0],
    N_samples=64,
    N_rand=2048,
    nerf_mode="image",
    depth_supervise=False,
    use_nerf_mask=True,
    nerf_sample_view=20,
    squeeze_scale=4,
    nerf_density=True,
    )
train_cfg = dict()
test_cfg = dict(
    nms_pre=1000,
    nms_thr=.15,
    use_rotate_nms=True,
    score_thr=.0)
img_norm_cfg = dict(mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
input_modality = dict(
    use_image=True,
    use_depth=False,
    use_lidar=False,
    use_neuralrecon_depth=False,
    use_ray=True)

train_collect_keys = ['img', 'gt_bboxes_3d', 'gt_labels_3d']
test_collect_keys = ['img']
if input_modality['use_depth']:
    train_collect_keys.append('depth')
    test_collect_keys.append('depth')
if input_modality['use_lidar']:
    train_collect_keys.append('lidar')
    test_collect_keys.append('lidar')
if input_modality['use_ray']:
    for key in [
        # 'c2w',
        # 'camrotc2w',
        'lightpos',
        # 'pixels',
        'nerf_sizes',
        'raydirs',
        'gt_images',
        'gt_depths',
        'denorm_images'
    ]:
        train_collect_keys.append(key)
        test_collect_keys.append(key)
        
dataset_type = 'ARKitMultiViewDataset'
data_root = 'data/arkit/'
class_names = ("cabinet", "refrigerator", "shelf", "stove", "bed", "sink", "washer", "toilet", "bathtub", "oven", "dishwasher", "fireplace", "stool", "chair", "table", "tv_monitor", "sofa")



train_pipeline = [
    dict(type='LoadAnnotations3D'),
    dict(
        type='MultiViewPipeline',
        n_images=50,
        transforms=[
            dict(type='LoadImageFromFile'),
            dict(type='Resize', img_scale=(320, 240), keep_ratio=True),
            dict(type='Normalize', **img_norm_cfg),
            dict(type='Pad', size=(240, 320))
        ],
        mean = [123.675, 116.28, 103.53],
        std = [58.395, 57.12, 57.375],
        margin = 10,
        depth_range=[0.5, 5.5],
        loading='random',
        nerf_target_views=10,
        ),
    dict(type='RandomShiftOrigin', std=(.7, .7, .0)), # this may lead to some issues in nerf.
    dict(type='DefaultFormatBundle3D', class_names=class_names),
    dict(type='Collect3D', keys=train_collect_keys)
]

test_pipeline = [
    dict(
        type='MultiViewPipeline',
        n_images=101,
        transforms=[
            dict(type='LoadImageFromFile'),
            dict(type='Resize', img_scale=(320, 240), keep_ratio=True),
            dict(type='Normalize', **img_norm_cfg),
            dict(type='Pad', size=(240, 320))
        ],
        mean = [123.675, 116.28, 103.53],
        std = [58.395, 57.12, 57.375],
        margin = 10,
        depth_range=[0.5, 5.5],
        loading="random",
        nerf_target_views=1,
        ),
    dict(type='DefaultFormatBundle3D', class_names=class_names, with_label=False),
    dict(type='Collect3D', keys=test_collect_keys)
]

data = dict(
    samples_per_gpu=1,
    workers_per_gpu=1,
    train=dict(
        type='RepeatDataset',
        times=2,
        dataset=dict(
            type=dataset_type,
            data_root=data_root,
            ann_file=data_root + 'arkit_infos_train.pkl',
            pipeline=train_pipeline,
            modality=input_modality,
            classes=class_names,
            filter_empty_gt=True,
            box_type_3d='Depth')),
    val=dict(
        type=dataset_type,
        data_root=data_root,
        ann_file=data_root + 'arkit_infos_val.pkl',
        pipeline=test_pipeline,
        modality=input_modality,
        classes=class_names,
        test_mode=True,
        box_type_3d='Depth'),
    test=dict(
        type=dataset_type,
        data_root=data_root,
        ann_file=data_root + 'arkit_infos_val.pkl',
        pipeline=test_pipeline,
        modality=input_modality,
        classes=class_names,
        test_mode=True,
        box_type_3d='Depth')
)

optimizer = dict(
    type='AdamW',
    lr=0.0002,
    weight_decay=0.0001,
    paramwise_cfg=dict(
        custom_keys={'backbone': dict(lr_mult=0.1, decay_mult=1.0)}))
optimizer_config = dict(grad_clip=dict(max_norm=35., norm_type=2))
lr_config = dict(policy='step', step=[8, 11])
total_epochs = 12

checkpoint_config = dict(interval=1)
log_config = dict(
    interval=50,
    hooks=[
        dict(type='TextLoggerHook'),
        dict(type='TensorboardLoggerHook')
    ])
evaluation = dict(interval=1)
dist_params = dict(backend='nccl')
find_unused_parameters = True  # todo: fix number of FPN outputs
log_level = 'INFO'
load_from = None
resume_from = '/home/sgl/NeRFDet/work_dirs/nerfdet_arkit/epoch_7.pth'
workflow = [('train', 1)]

**Comment by SerCharles** [comment]
Above is the config of training nerfdet on arkit

**Comment by SerCharles** [comment]
you can give me your mailbox，And i will send all the code to you

**Comment by SerCharles** [comment]
The data processing and running procedure of ARKit is almost the same as ScanNet. Since I did not write a README for running the baseline, and the experiment was done half a year ago, I do not remember exactly how to run it as well, just follow the procedure of ScanNet, and change the config.

**Comment by LIZECHUAN** [comment]
Thank you for your excellent work. Possible to share the ARKitMultiViewDataset and any other code related to training NeRFDet on ARKit with me? My email is: zechuanl@student.unimelb.edu.au 
Thank you very much!

**Comment by SerCharles** [comment]
在mmdet3d/datasets/arkit_dataset.py


| |
沈冠霖
| |
|
清华大学软件学院
***@***.***
|
---- Replied Message ----
| From | li ***@***.***> |
| Date | 8/15/2024 11:26 |
| To | ***@***.***> |
| Cc | Guanlin ***@***.***> ,
***@***.***> |
| Subject | Re: [SerCharles/CN-RMA] Possible to share your code of training nerfdet on arkit? (Issue #2) |

Thank you for your excellent work. Possible to share the ARKitMultiViewDataset and any other code related to training NeRFDet on ARKit with me? My email is: ***@***.***
Thank you very much!

**Comment by LIZECHUAN** [comment]
> 在mmdet3d/datasets/arkit_dataset.py | | 沈冠霖 | | | 清华大学软件学院 ***@***.*** | ---- Replied Message ---- | From | li ***@***.***> | | Date | 8/15/2024 11:26 | | To | ***@***.***> | | Cc | Guanlin ***@***.***> , ***@***.***> | | Subject | Re: [SerCharles/CN-RMA] Possible to share your code of training nerfdet on arkit? (Issue #2) | Thank you for your excellent work. Possible to share the ARKitMultiViewDataset and any other code related to training NeRFDet on ARKit with me? My email is: ***@***.*** Thank you very much! — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***>

感谢您的回复！好像没有呀，您现在代码好像没有mmdet3d这个目录
